src/main/java/frc/robot/subsystems/motorSubsystem/MotorCommandBuilder.java

```diff
@@ -0,0 +1,29 @@
+package frc.robot.subsystems.motorSubsystem;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import frc.utils.utilcommands.ExecuteEndCommand;
+
+import java.util.function.DoubleSupplier;
+
+public class MotorCommandBuilder {
+
+	private MotorSubsystem motor;
+
+	public MotorCommandBuilder(MotorSubsystem motor) {
+		this.motor = motor;
+	}
+
+	public Command setPower(DoubleSupplier powerSupplier) {
+		return new ExecuteEndCommand(() -> motor.setPower(powerSupplier.getAsDouble()), () -> {}, motor);
+	}
+
+	public Command setPower(double power) {
```

Comment:
RunCommand executes the Runnable in loop this and the function below can use an InstantCommand
the function above can use RunCommand

Reply:
also all three commands should require the subsystem